Update for ESLint 2.

nosilleg opened the pull request:
Also removes need for lodash merge.

nosilleg committed:
Update for ESLint 2.

Andersos commented on env-es6-modules.js:
```diff
@@ -1,12 +1,11 @@
-    'ecmaFeatures': {
-        'modules': true
+module.exports = {
+    'entends': [
```
Should be "extends"

nosilleg replied:
Updated.

Andersos commented:
I like the way you changed the lodash dependency for extends. Great work =)

nosilleg commented:
Re:lodash - Yeah, I don't know why it didn't occur to me when I wrote this originally. I guess it was the circular reference.

Andersos commented:
I didn't have push permissions to your repo so might as well post the changes here:
https://github.com/schibsted/eslint-config-spt/pull/3/files#diff-168726dbe96b3ce427e7fedce31bb0bcR14
This line is talking about this rule 'no-unexpected-multiline': [1, {"max": 2}],
I found nowhere in the [documentation](http://eslint.org/docs/rules/no-unexpected-multiline) that this rule can have extra arguments and I think that is why we are seeing the [errors](https://travis-ci.org/schibsted/eslint-config-spt/builds/119233041#L241) in Travis. 

Delete the empty line below and change the original line to this:
'no-unexpected-multiline': 2,

At that time the original problem is fixed but a new one is emerged. 
I think we are back to your comment in https://github.com/schibsted/eslint-config-spt/pull/3#issuecomment-202877347 about the circular references and that the `extends` can't be used because of that =( 

```bash
Cannot find module 'eslint-config-spt/env-node-false' from '/Users/anders/dev/eslint-config-spt/node_modules'
Referenced from: /Users/anders/dev/eslint-config-spt/index.js
Error: Cannot find module 'eslint-config-spt/env-node-false' from '/Users/anders/dev/eslint-config-spt/node_modules'
Referenced from: /Users/anders/dev/eslint-config-spt/index.js
    at Function.module.exports [as sync] (/Users/anders/dev/eslint-config-spt/node_modules/resolve/lib/sync.js:33:11)
    at resolve (/Users/anders/dev/eslint-config-spt/node_modules/eslint/lib/config/config-file.js:442:38)
    at load (/Users/anders/dev/eslint-config-spt/node_modules/eslint/lib/config/config-file.js:462:24)
    at /Users/anders/dev/eslint-config-spt/node_modules/eslint/lib/config/config-file.js:369:36
    at Array.reduceRight (native)
    at applyExtends (/Users/anders/dev/eslint-config-spt/node_modules/eslint/lib/config/config-file.js:346:28)
    at Object.load (/Users/anders/dev/eslint-config-spt/node_modules/eslint/lib/config/config-file.js:491:22)
    at loadConfig (/Users/anders/dev/eslint-config-spt/node_modules/eslint/lib/config.js:67:33)
    at new Config (/Users/anders/dev/eslint-config-spt/node_modules/eslint/lib/config.js:202:38)
    at CLIEngine.executeOnFiles (/Users/anders/dev/eslint-config-spt/node_modules/eslint/lib/cli-engine.js:493:28)
npm ERR! Test failed.  See above for more details.
```

Instead of dragging in lodash again may I suggest we try using [Object.assign](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign).

nosilleg commented:
Oops.  Updated `no-unexpected-multiline` but meant to update `no-multiline-empty-lines` instead.  I've updated and reset `no-unexpected-multiline` back to 1.

As for the circular reference, this is only a problem for `npm test` since there isn't a local version of the `eslint-config-spt` package in `node_modules`.  It's not actually a problem when the package is installed into another project.  

We could update to go the old way (but with Object.assign); make the package a dependency of itself (which would then be one version out of date); or as part of `npm test` copy the package into  `node_modules` thereby satisying the dependency and using the current version.

That last one is probably my preferred.  Thoughts?

Andersos commented:
LGTM

A note on the symlink. We decided to add a file in the `node_modules` folder that symlinks back to the parent directly to make it easier to use the same rules to lint this project.
